firebase-app-distribution/src/main/java/com/google/firebase/appdistribution/UpdateTaskImpl.java

```diff
@@ -0,0 +1,123 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.appdistribution;
+
+import android.app.Activity;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import com.google.android.gms.tasks.OnFailureListener;
+import com.google.android.gms.tasks.OnSuccessListener;
+import com.google.android.gms.tasks.Task;
+import java.util.concurrent.Executor;
+
+/** Implementation of UpdateTask, the return type of updateApp. */
+class UpdateTaskImpl extends UpdateTask {
+
+  @NonNull private final Task<UpdateState> task;
+  @Nullable private OnProgressListener listener = null;
+
+  public UpdateTaskImpl(@NonNull Task<UpdateState> task) {
+    this.task = task;
+  }
+
+  public void updateProgress(@NonNull UpdateState updateState) {
```

Comment:
should this be public? Do we want customers to be calling this method from their code? Perhaps it should be package private?

Reply:
Yep I think package private makes more sense